### ISSUE TITLE
Is convolution.convolve_fft(nan_treatment='fill') implemented?

### ISSUE THREAD

**Issue by jamienoss** [astropy/astropy#8121]
``convolve_fft()`` docstring [[src](https://github.com/astropy/astropy/blob/master/astropy/convolution/convolve.py#L434-L442)]
```python
"""
fill_value : float, optional
        The value to use outside the array when using boundary='fill'
nan_treatment : 'interpolate', 'fill'
        ``interpolate`` will result in renormalization of the kernel at each
        position ignoring (pixels that are NaN in the image) in both the image
        and the kernel.  ``fill`` will replace the NaN pixels with a fixed
        numerical value (default zero, see ``fill_value``) prior to
        convolution.  Note that if the kernel has a sum equal to zero, NaN
        interpolation is not possible and will raise an exception.
"""
```
(minor) The ``nan_treatment`` doc implies that ``fill_value`` is utilized as that for ``nan_treatment='fill'`` as well as that for ``boundary='fill'``, however, this is not reflected in doc for ``fill_value``.

(major) ``fill_value`` doesn't seem to be used for ``nan_treatment='fill'``. I can't find the code.
(major) I can't find the code for the above because I can't even find the implementation for ``nan_treatment='fill'``. I don't think it is implemented.

@keflavich would you mind providing links for the relevant sections, please? Thanks.

To me, it looks like this section might perhaps be the implementation however, it, unfortunately, doesn't implement ``nan_treatment='fill'``, so maybe it isn't.

[[src](https://github.com/astropy/astropy/blob/master/astropy/convolution/convolve.py#L754-L771)]
```python
...
interpolate_nan = (nan_treatment == 'interpolate')
    if interpolate_nan:
        if not np.isfinite(fill_value): # <---- this block looks like it deals with just the padding (like [here](https://github.com/astropy/astropy/blob/master/astropy/convolution/convolve.py#L735-L738))
            bigimwt = np.zeros(newshape, dtype=complex_dtype)
        else:
            bigimwt = np.ones(newshape, dtype=complex_dtype)

       bigimwt[arrayslices] = 1.0 - nanmaskarray * interpolate_nan # <--- attempt at filling
        wtfft = fftn(bigimwt)
...
    else:
          bigimwt = 1
...
```

1) It's conditional on ``nan_treatment == 'interpolate'`` and not ``'fill'`` so won't even be executed.
2) The ``fill_value`` isn't actually used.
3) ``bigimwt`` is only subsequently referenced within another block conditional on ``interpolate_nan`` ([src](https://github.com/astropy/astropy/blob/master/astropy/convolution/convolve.py#L788-L801)) and therefore the ``else: bigimwt = 1`` has no use.

There's no section of code that tethers ``nanmaskarray`` and ``fill_value`` for ``nan_treatment='fill'``.

I feel like I've missed something obvious but at the same time, am convinced that it's just not there.

**Comment by keflavich** [comment]
You're right.  A clause like this:
```
    if interpolate_nan == 'fill':
        rifft[arrayslices][nanmaskarray] = fill_value
```
just above `if preserve_nan` is needed.

I'm curious how this was missed.  Clearly we don't have a test for it.  I wonder if I just overlooked it when trying to match the convolve/convolve_fft parameters or if it was accidentally deleted somewhere in history...

**Comment by larrybradley** [comment]
Fixed in #8122.
